nixos/modules/programs/captive-browser.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.captive-browser;
+
+  package = pkgs.buildGoPackage rec {
+    name = "captive-browser";
+    version = "2018-11-08";
+    goPackagePath = name;
+
+    src = pkgs.fetchFromGitHub {
+      owner  = "FiloSottile";
+      repo   = "captive-browser";
+      rev    = "0f84599ddb36c090cf000681f86321aacb354c4a";
+      sha256 = "0fw4g4fsj9gwm711yj81z64bnmfj2g7wfkhnlsj1nfav0n5kfh40";
+    };
+  };
+in
+{
+  ###### interface
+
+  options = {
+    programs.captive-browser = {
+      enable = mkEnableOption "captive browser";
+
+      wlan-interface = mkOption {
+        type = types.str;
+        description = "your public network interface (wlp3s0, wlan0, eth0, ...)";
+      };
+
+      # the options below are the same as in "captive-browser.toml"
+      browser = mkOption {
+        type = types.str;
+        default = concatStringsSep " " [ ''${pkgs.chromium}/bin/chromium''
+                                         ''--user-data-dir=/tmp/.chromium-captive''
+                                         ''--proxy-server="socks5://$PROXY"''
+                                         ''--host-resolver-rules="MAP * ~NOTFOUND , EXCLUDE localhost"''
+                                         ''--no-first-run''
+                                         ''--new-window''
+                                         ''--incognito''
+                                         ''http://opennet.ru''
```

Comment:
I believe this should be `http://example.com` (it's a special reserved domain that works nicely for this scenario).

I think the premise is we don't want to leak data to a random server over unencrypted http while trying to get hijacked by a captive portal.

Reply:
https://github.com/NixOS/nixpkgs/blob/403de7911046751b6da298cdde1d30528c734183/nixos/modules/programs/captive-browser.nix#L55

In the airport I have tested this (Novosibirsk), ```example.com``` was resolving to ```127.0.0.1``` by the captive DNS server and thus the browser did not open the captive portal because made no network connection

Reply:
Ah missed that comment.

I'd still feel pretty uncomfortable with using a random .ru domain as the default. Maybe it's better to stick with `http://example.com/` and expose an option to override easily?

For your specific case, could also try example.org, example.net? For me personally, example.com has always worked.

Another option would be to use Apple's or Google's official captive portal URLs:
- http://captive.apple.com/hotspot-detect.html
- http://clients3.google.com/generate_204

Reply:
```http://cache.nixos.org``` may be a good default website here.

I hope it will never redirect to https (as http://nixos.org does) because it forcing https will break many scripts (for example, Perl's ```File::Fetch``` does not support https out of the box)

Reply:
That's a pretty cute compromise, I'm into it. :) :+1:

Reply:
Ok, I will change to ```cache.nixos.org```
```something.google.com``` might have better uptime, but isn't it blacklisted in China?

Reply:
Not sure, but that's what their devices use.

`cache.nixos.org` seems like a good starting point at least that people aren't likely to be offended by. :)

Reply:
There is also `http://detectportal.firefox.com/`